homeassistant/components/camera/mqtt.py

```diff
@@ -0,0 +1,71 @@
+"""
+Camera that loads a picture from an MQTT topic.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.mqtt/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.components.mqtt as mqtt
+from homeassistant.const import CONF_NAME
+from homeassistant.components.camera import Camera, PLATFORM_SCHEMA
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TOPIC = 'topic'
+
+DEFAULT_NAME = 'MQTT Camera'
+
+DEPENDENCIES = ['mqtt']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOPIC): mqtt.valid_subscribe_topic,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Camera."""
+    topic = config[CONF_TOPIC]
+
+    add_devices([MqttCamera(config[CONF_NAME], topic)])
+
+
+class MqttCamera(Camera):
+    """MQTT camera."""
+
+    def __init__(self, name, topic):
+        """Initialize Local File Camera component."""
+        super().__init__()
+
+        self._name = name
+        self._topic = topic
+        self._qos = 0
+        self._last_image = None
+
+    def camera_image(self):
```

Comment:
Thanks Pascal, I noticed that a few requested changes are on stuff I just copied from other components. In this case, for example, from the local_file camera.
I understand that new directives are in place and that not all the old code has been revised yet. Is there a way to know what's changed, in advance? It would save some time...